api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -547,6 +547,9 @@ public String getDrugNonCoded() {
 	 */
 	public void setDrugNonCoded(String drugNonCoded) {
 		this.drugNonCoded = StringUtils.isNotBlank(drugNonCoded) ? drugNonCoded.trim() : drugNonCoded;
+		if (StringUtils.isNotBlank(drugNonCoded) && getConcept() == null) {
```

Comment:
Do you need this if you are already setting it at the service layer in OrderServiceImpl?

Reply:
I think we need it in both `DrugOrder` and `OrderServiceImpl`